sc/src/main/java/uk/ac/ebi/atlas/tsne/TSnePlotService.java

```diff
@@ -29,20 +29,15 @@ public TSnePlotService(TSnePlotServiceDao tSnePlotServiceDao, CellMetadataDao ce
 
     public Set<TSnePoint> fetchTSnePlotWithExpression(String experimentAccession, int perplexity, String geneId) {
         return tSnePlotServiceDao.fetchTSnePlotWithExpression(experimentAccession, perplexity, geneId).stream()
-                //.filter(line -> Math.round(line.x())%2==0)
-                //.filter(line -> Math.round(line.y())%2==0)
                 .map(
                         pointDto ->
                                 TSnePoint.create(
-                                        pointDto.x(), pointDto.y(), pointDto.expressionLevel(), pointDto.name()))
+                                       Math.round(pointDto.x()*100.0)/100.0 ,Math.round(pointDto.y()*100.0)/100.0, pointDto.expressionLevel(), pointDto.name()))
```

Comment:
We use [Apache Commons MathUtils](https://commons.apache.org/proper/commons-math/javadocs/api-2.2/org/apache/commons/math/util/MathUtils.html) for this sort of numerical manipulation. Java’s `Math.round` is quite poor. The alternative is converting to `BigDecimal` but MathUtils saves us all that trouble.

Also, remember that you should add a test case to `TSnePlotServiceDaoIT` to verify the correctness of your code.